pyproject.toml

```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
+name = "streamlit-aibrary"
+version = "0.0.1"
+description = ""
+authors = ["Alireza Morsali <alireza.mors@gmail.com>"]
+package-mode = false
+
+[tool.poetry.dependencies]
+python = "^3.12"
+openai = "^1.6.1"
+streamlit = "^1.41.1"
+langchain-community = "^0.3.14"
+faiss-cpu = "^1.9.0.post1"
+markitdown = "^0.0.1a3"
+tiktoken = "^0.8.0"
+langchain-openai = "^0.3.0"
+jupyter = "^1.0.0"
+black = "^23.9.1"
+isort = "^5.12.0"
+pytest = "^7.4.4"
+pytest-asyncio = "^0.23.6"
+python-dotenv = "^1.0.1"
+aibrary = "^0.0.2"
```

Comment:
aibrary installs many of these packages as its own dependencies.
Can you only keep the ones that are needed here.